enterprise/cli/dbcrypt_rotate.go

```diff
@@ -0,0 +1,152 @@
+			var newKey []byte
+			for idx, ek := range vals.ExternalTokenEncryptionKeys {
+				dk, err := base64.StdEncoding.DecodeString(ek)
+				if err != nil {
+					return xerrors.Errorf("key must be base64-encoded")
+				}
+				if idx == 0 {
+					newKey = dk
+				} else if bytes.Equal(dk, newKey) {
+					return xerrors.Errorf("old key at index %d is the same as the new key", idx)
+				}
+				keys = append(keys, dk)
+			}
+
+			ciphers, err := dbcrypt.NewCiphers(keys...)
+			if err != nil {
+				return xerrors.Errorf("create ciphers: %w", err)
+			}
+
+			sqlDB, err := cli.ConnectToPostgres(inv.Context(), logger, "postgres", vals.PostgresURL.Value())
+			if err != nil {
+				return xerrors.Errorf("connect to postgres: %w", err)
+			}
+			defer func() {
+				_ = sqlDB.Close()
+			}()
+			logger.Info(ctx, "connected to postgres")
+
+			db := database.New(sqlDB)
+
+			cryptDB, err := dbcrypt.New(ctx, db, ciphers...)
+			if err != nil {
+				return xerrors.Errorf("create cryptdb: %w", err)
+			}
+
+			users, err := cryptDB.GetUsers(ctx, database.GetUsersParams{})
+			if err != nil {
+				return xerrors.Errorf("get users: %w", err)
+			}
+			logger.Info(ctx, "encrypting user tokens", slog.F("count", len(users)))
+			for idx, usr := range users {
+				err := cryptDB.InTx(func(tx database.Store) error {
+					userLinks, err := tx.GetUserLinksByUserID(ctx, usr.ID)
+					if err != nil {
+						return xerrors.Errorf("get user links for user: %w", err)
+					}
+					for _, userLink := range userLinks {
+						if _, err := tx.UpdateUserLink(ctx, database.UpdateUserLinkParams{
+							OAuthAccessToken:  userLink.OAuthAccessToken,
+							OAuthRefreshToken: userLink.OAuthRefreshToken,
+							OAuthExpiry:       userLink.OAuthExpiry,
+							UserID:            usr.ID,
+							LoginType:         usr.LoginType,
+						}); err != nil {
+							return xerrors.Errorf("update user link: %w", err)
+						}
+					}
+					gitAuthLinks, err := tx.GetGitAuthLinksByUserID(ctx, usr.ID)
+					if err != nil {
+						return xerrors.Errorf("get git auth links for user: %w", err)
+					}
+					for _, gitAuthLink := range gitAuthLinks {
+						if _, err := tx.UpdateGitAuthLink(ctx, database.UpdateGitAuthLinkParams{
+							ProviderID:        gitAuthLink.ProviderID,
+							UserID:            usr.ID,
+							UpdatedAt:         gitAuthLink.UpdatedAt,
+							OAuthAccessToken:  gitAuthLink.OAuthAccessToken,
+							OAuthRefreshToken: gitAuthLink.OAuthRefreshToken,
+							OAuthExpiry:       gitAuthLink.OAuthExpiry,
+						}); err != nil {
+							return xerrors.Errorf("update git auth link: %w", err)
+						}
+					}
+					return nil
+				}, &sql.TxOptions{
+					Isolation: sql.LevelRepeatableRead,
+				})
+				if err != nil {
+					return xerrors.Errorf("update user links: %w", err)
+				}
+				logger.Debug(ctx, "encrypted user tokens", slog.F("current", idx+1), slog.F("cipher", ciphers[0].HexDigest()))
+			}
+			logger.Info(ctx, "operation completed successfully")
```

Comment:
Do you think that we need to perform a separate sanity check to make sure that the operation went smoothly?

Reply:
Similar to what we do in the associated unit tests, we could re-read the affected user(s) and validate that the required fields are encrypted with the correct cipher. Does that sound OK to you?